### PR TITLE
profile page (fixes #402)

### LINKED ISSUE #402: Use same component for the profile page(community members and user profile page)

### PR DESCRIPTION
**Pull request by SujalLama**
Fixes #402 

### Description
Single component for myprofile and communitymemberprofile
Delete member profile component
back button works now
Edit button for my profile and follow button for community member

### Links and Curls
deleted:
src/screens/communityMemberProfile/CommunityMemberProfile.jsx
src/screens/communityMemberProfile/CommunityMemberProfile.scss
src/screens/dashboard/MyProfile.jsx
src/screens/dashboard/MyProfile.scss

modified:
src/App.jsx
src/screens/dashboard/MainDashboard.jsx
src/screens/dashboard/MyProfile.jsx
api/src/controllers/communityUserController.js
api/src/controllers/userController.js
src/actions/userAction.js
src/components/cardImage/CardImage.jsx

added:
src/screens/profile/

routes: 
http://localhost:3000/community-members-profile/2
http://localhost:3000/myProfile
These above routes are changed to:
http://localhost:3000/profile/:userID

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -77,7 +76,7 @@ function App () {
             <PrivateRoute component={CommunityNewsViewPage} path='/community-page-news-view' exact />
             <PrivateRoute component={AllCommunitiesCard} path='/community-switching' exact />
             <PrivateRoute component={CommunityMembers} path='/community-members/:id' exact />
-            <PrivateRoute component={CommunityMembersProfile} path='/community-members-profile/:id' exact />
+            <PrivateRoute component={MyProfile} path='/community-members-profile/:id' exact />
```
Comment (lmmrssa) [inline src/App.jsx:79]: url should be generic as it is general profile page not specific to community

File: src/screens/dashboard/MyProfile.jsx [hunk 2]
```diff
@@ -8,19 +8,24 @@ import {
...
+  const { pathname } = useLocation()
   const [showEmailVerificationModal, setShowEmailVerificationModal] = useState(false)
   const [showPhoneVerificationModal, setShowPhoneVerificationModal] = useState(false)
   const history = useHistory()
   const dispatch = useDispatch()
   const userDetails = useSelector((state) => state.userDetails)
   const { status } = useSelector((state) => state.userAttrConfirmCode)
   const { user, loading } = userDetails
+  const userLogin = useSelector((state) => state.userLogin)
+  const { userInfo } = userLogin
```
Comment (lmmrssa) [inline src/screens/dashboard/MyProfile.jsx:28]: this is current logged in user and not sure this is required as it was not being sued before and getMyDetails was getting details details for user profile

File: src/screens/dashboard/MyProfile.jsx [hunk 3]
```diff
@@ -62,19 +78,31 @@ function MyProfile () {
             showPhoneVerificationModal &&
               <VerificationModal type='phone' clickHandler={phoneClickHandler} />
             }
-            <DashboardLayout title='My Profile'>
+            <DashboardLayout
+              title={
+              pathname === '/myProfile'
+                ? 'My Profile'
+                : 'Community Members'
```
Comment (lmmrssa) [inline src/screens/dashboard/MyProfile.jsx:85]: This is actually User profile not community members

File: src/screens/dashboard/MyProfile.jsx [hunk 4]
```diff
@@ -62,19 +78,31 @@ function MyProfile () {
...
+                        <PersonalInformation user={user} />
+                        {user?.phone === undefined || <ContactInformation user={user} />}
+                        <AdditionalInformation user={user} />
+                      </>}
                     </div>
-                    <EditInformation clickHandler={editUserInformation} image={user?.attachments && process.env.REACT_APP_CDN_BASE_URL + '/attachments/' + user.attachments} />
+                    <EditInformation
+                      clickHandler={editUserInformation}
+                      image={user?.attachments && process.env.REACT_APP_CDN_BASE_URL + '/attachments/' + user.attachments}
+                      follow={pathname !== '/myProfile'}
```
Comment (lmmrssa) [inline src/screens/dashboard/MyProfile.jsx:104]: follow should not show if logged in user and current profile is not same irrespective of url

**Review by lmmrssa — CHANGES_REQUESTED** [review]

**Review by lmmrssa — APPROVED** [review]

**Comment by dogi** [comment]
image
removing phone number -> undefined

**Review by dogi — APPROVED** [review]

**Pull request merged**